hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1669,21 +1670,27 @@ public void start() throws IOException {
     }
 
     // Set metrics and start metrics back ground thread
-    metrics.setNumVolumes(metadataManager.countRowsInTable(metadataManager
-        .getVolumeTable()));
-    metrics.setNumBuckets(metadataManager.countRowsInTable(metadataManager
-        .getBucketTable()));
-
-    if (getMetricsStorageFile().exists()) {
-      OmMetricsInfo metricsInfo = READER.readValue(getMetricsStorageFile());
-      metrics.setNumKeys(metricsInfo.getNumKeys());
-    }
+    CompletableFuture.runAsync(() -> {
+      try {
+        metrics.setNumVolumes(
+            metadataManager.countRowsInTable(metadataManager.getVolumeTable()));
+        metrics.setNumBuckets(
+            metadataManager.countRowsInTable(metadataManager.getBucketTable()));
+
+        if (getMetricsStorageFile().exists()) {
+          OmMetricsInfo metricsInfo = READER.readValue(getMetricsStorageFile());
+          metrics.setNumKeys(metricsInfo.getNumKeys());
+        }
 
-    // FSO(FILE_SYSTEM_OPTIMIZED)
-    metrics.setNumDirs(metadataManager
-        .countEstimatedRowsInTable(metadataManager.getDirectoryTable()));
-    metrics.setNumFiles(metadataManager
-        .countEstimatedRowsInTable(metadataManager.getFileTable()));
+        // FSO(FILE_SYSTEM_OPTIMIZED)
+        metrics.setNumDirs(metadataManager.countEstimatedRowsInTable(
+            metadataManager.getDirectoryTable()));
+        metrics.setNumFiles(metadataManager.countEstimatedRowsInTable(
+            metadataManager.getFileTable()));
+      } catch (IOException e) {
+        LOG.warn("Async set om metrics fail.", e);
+      }
+    });
```

Comment:
It may be necessary to continue to throw the Exception here.

Reply:
> It may be necessary to continue to throw the Exception here.

Thank you for the review. I have updated the code to throw the Exception here.